### ISSUE TITLE
bug: delegatesFocus patch for ios causes rendering issues with item sliding in chrome

### ISSUE THREAD

**Issue by deving1995** [ionic-team/ionic-framework#25273]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When you click to drag an ion-item inside of an ion-item-sliding and you have an ion-input in the ion-item if you start the drag by hovering over the ion-label, then end the drag while hovering over the ion-input the slider abruptly jiggles to the left, slams shut, and when you drag it to open it again the ion-item-option is missing.

You can fix the ion-item-sliding by dragging it back to the left even though it is already shut.

I believe the issue is that the ion-item-option is somehow being pushed out of view.

### Expected Behavior

I expect that when I am intending to drag an ion-item inside an ion-item-sliding that releasing the mouse while hovering over an ion-input will not cause the ion-item-sliding to close, or for the ion-item-option to disappear when re-opening.

### Steps to Reproduce

1. Click the `ion-label` of the `ion-item`
    - Do NOT let go of the mouse
2. Drag the `ion-item` to the right until the cursor turns into a text-cursor
3. Release the mouse
4. Open the `ion-item-sliding` back up to see that the `ion-item-option` is now gone

```html
<ion-item-sliding>
  <ion-item-options side="start">
    <ion-item-option>Test</ion-item-option>
  </ion-item-options>
  <ion-item>
    <ion-label>Test Slider</ion-label>
    <ion-input></ion-input>
  </ion-item>
</ion-item-sliding>
```

Before option disappears
![image](https://user-images.githubusercontent.com/87441052/167725116-90260e5d-672f-47c6-827e-b3cf55cef9a3.png)

After option disappears
![image](https://user-images.githubusercontent.com/87441052/167725288-df2274ec-f892-40e8-82a9-c2ef79ecdb61.png)

And here's a gif of the issue in action
![slider-issue](https://user-images.githubusercontent.com/87441052/167725577-ceaacc95-151f-4851-bc65-2a50bf95da21.gif)

### Code Reproduction URL

https://github.com/deving1995/ion-item-sliding-issue

### Ionic Info

[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package'

       Require stack:
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\lib\project\index.js
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\lib\index.js
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\index.js
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\bin\ionic
[WARN] Error loading @capacitor/android package.json: Error: Cannot find module
       '@capacitor/android/package'

       Require stack:
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\lib\project\index.js
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\lib\index.js        
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\index.js
       - C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli\bin\ionic

Ionic:

   Ionic CLI                     : 6.18.1 (C:\Users\dgero\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.1.4
   @angular-devkit/build-angular : 13.2.6
   @angular-devkit/schematics    : 13.2.6
   @angular/cli                  : 13.2.6
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.5.1
   @capacitor/android : not installed
   @capacitor/core    : 3.5.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.5.0

System:

   NodeJS : v14.17.1 (C:\Program Files\nodejs\node.exe)
   npm    : 8.6.0
   OS     : Windows 10

### Additional Information

The reason I believe the issue is that the `ion-item-option` is being pushed off screen is because you can actually get the `ion-item-option` to be partially out of view.

![slider-issue2](https://user-images.githubusercontent.com/87441052/167724494-35f22229-b652-4400-b4b9-8b9a3a39f651.png)
.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This appears to be a Chrome specific bug. Focusing the input seems to cause the view to scroll/translate.

**Comment by deving1995** [comment]
Thank you for the information, is there a workaround for this?

**Comment by liamdebeasi** [comment]
This issue is caused by https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/item/item.tsx#L294. There is a bug on iOS 13 where [delegatesFocus](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/delegatesFocus) is not working correctly. As a result, we added a patch to try and fix the behavior.

Unfortunately, this had the unintended side effect of introducing the behavior reported here. Once we drop support for iOS 13, we should be able to remove this patch.

-----

As for a workaround, you could try using a native `<input>` element instead of `<ion-input>` for now. The patch only activates if `ion-item` detects an `ion-input` or `ion-textarea`: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/item/item.tsx#L284

**Comment by liamdebeasi** [comment]
Hi there,

We are proposing some changes to form components that seek to resolve this issue. We would love your feedback on these proposed changes! You can read more about the changes and leave comments here: https://github.com/ionic-team/ionic-framework/discussions/25661

**Comment by deving1995** [comment]
Thank you!

**Comment by liamdebeasi** [comment]
Hey there,

I merged a fix in via https://github.com/ionic-team/ionic-framework/pull/25822 that resolves this issue. The relevant code was originally there to work around an iOS 13 bug. We will be dropping iOS 13 support in Ionic 7, so we are targeting the fix for that release.

I am going to close this since the code has been merged, but let me know if you have any questions. Thanks for the report!
